state/binarystorage/binarystorage.go

```diff
@@ -136,62 +135,50 @@ func (s *toolsStorage) Tools(v version.Binary) (Metadata, io.ReadCloser, error)
 		Size:    metadataDoc.Size,
 		SHA256:  metadataDoc.SHA256,
 	}
-	return metadata, tools, nil
+	return metadata, r, nil
 }
 
-func (s *toolsStorage) Metadata(v version.Binary) (Metadata, error) {
-	metadataDoc, err := s.toolsMetadata(v)
+func (s *binaryStorage) Metadata(version string) (Metadata, error) {
```

Comment:
Perhaps consider leaving "version" out of this and calling it "name", as there's nothing in this package that relies on the fact it's actually a version AFAICS.